Marlin/src/HAL/HAL_AVR/fastio_AVR.h

```diff
@@ -65,26 +65,24 @@
 
 #define _READ(IO) ((bool)(DIO ## IO ## _RPORT & _BV(DIO ## IO ## _PIN)))
 
-// On some boards pins > 0x100 are used. These are not converted to atomic actions. A critical section is needed.
+// On some boards pins > 0x100 are used. These are not converted to atomic actions. We need to use
+// the atomic bit flipping capability of the port
 
 #define _WRITE_NC(IO, v)  do { if (v) {DIO ##  IO ## _WPORT |= _BV(DIO ## IO ## _PIN); } else {DIO ##  IO ## _WPORT &= ~_BV(DIO ## IO ## _PIN); }; } while (0)
 
-#define _WRITE_C(IO, v)   do { if (v) { \
-                                         CRITICAL_SECTION_START; \
-                                         {DIO ##  IO ## _WPORT |= _BV(DIO ## IO ## _PIN); } \
-                                         CRITICAL_SECTION_END; \
-                                       } \
-                                       else { \
-                                         CRITICAL_SECTION_START; \
-                                         {DIO ##  IO ## _WPORT &= ~_BV(DIO ## IO ## _PIN); } \
-                                         CRITICAL_SECTION_END; \
-                                       } \
-                                     } \
-                                     while (0)
+#define _WRITE_C(IO, v) do { \
+  /* Read port to get the current state of each pin of the port */ \
+  uint8_t changeMask = DIO ##  IO ## _WPORT; \
+  /* Compute a bitmask with the pin value we want to change */ \
+  if (v) changeMask = ~changeMask; \
+  changeMask &= _BV(DIO ## IO ## _PIN); \
+  /* Use atomic flipping of output port bits */ \
+  DIO ##  IO ## _RPORT = changeMask; \
+} while (0)
```

Comment:
I guess this change will only affect platforms that have a Port H and above.
Which AVR platforms have that many ports?